### PR TITLE
Created Chunk save and load event. Closes #89

### LINKED ISSUE #89: Requesting ChunkDataEvent (Original from Forge)
I'd like to request the ChunkDataEvent.Save and ChunkDataEvent.Load bridging for architectury - It is a very important event for any mod that stores world data.

### PR DESCRIPTION
**Pull request by canitzp**

### CODE REVIEW

**Comment by MaxNeedsSnacks** [comment]
I feel like these should have the suffix `_DATA` as to not confuse them with actual chunk loading / unloading (which may also need an event down the line)

File: common/src/main/java/me/shedaniel/architectury/event/events/ChunkEvent.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+         * @param nbt The chunk data that is written to the save file.
+         */
+        void save(ChunkAccess chunk, ServerLevel level, CompoundTag nbt);
+    }
+    
+    interface Load {
+        /**
+         * Invoked just before a chunk is fully read.
+         * You can read out your own data from the {@link CompoundTag} parameter, when you have saved one before.
+         * Equal to Forge's {@code ChunkDataEvent.Load}.
```
Comment (shedaniel) [inline common/src/main/java/me/shedaniel/architectury/event/events/ChunkEvent.java]:
```suggestion
         * Equivalent to Forge's {@code ChunkDataEvent.Load}.
```

File: common/src/main/java/me/shedaniel/architectury/event/events/ChunkEvent.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    
+    Event<Save> SAVE = EventFactory.createLoop();
+    
+    Event<Load> LOAD = EventFactory.createLoop();
+    
+    interface Save {
+        /**
+         * Invoked when a chunk is saved, just before the data is written.
+         * Add your own data to the {@link CompoundTag} parameter to get your data saved as well.
+         * Equal to Forge's {@code ChunkDataEvent.Save}.
```
Comment (shedaniel) [inline common/src/main/java/me/shedaniel/architectury/event/events/ChunkEvent.java]:
```suggestion
         * Equivalent to Forge's {@code ChunkDataEvent.Save}.
```

File: common/src/main/java/me/shedaniel/architectury/event/events/ChunkEvent.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        /**
+         * Invoked just before a chunk is fully read.
+         * You can read out your own data from the {@link CompoundTag} parameter, when you have saved one before.
+         * Equal to Forge's {@code ChunkDataEvent.Load}.
+         *
+         * @param chunk The chunk that is loaded.
+         * @param type The type of the chunk.
+         * @param nbt THe chunk data that was read from the save file.
+         */
+        void load(ChunkAccess chunk, ChunkStatus.ChunkType type, CompoundTag nbt);
```
Comment (shedaniel) [inline common/src/main/java/me/shedaniel/architectury/event/events/ChunkEvent.java]: Is the ServerLevel not available here?

File: common/src/main/java/me/shedaniel/architectury/event/events/ChunkEvent.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    
+    interface Load {
+        /**
+         * Invoked just before a chunk is fully read.
+         * You can read out your own data from the {@link CompoundTag} parameter, when you have saved one before.
+         * Equal to Forge's {@code ChunkDataEvent.Load}.
+         *
+         * @param chunk The chunk that is loaded.
+         * @param type The type of the chunk.
+         * @param nbt THe chunk data that was read from the save file.
```
Comment (shedaniel) [inline common/src/main/java/me/shedaniel/architectury/event/events/ChunkEvent.java]:
```suggestion
         * @param nbt The chunk data that was read from the save file.
```

**Review by shedaniel — APPROVED** [review]

**Review by MaxNeedsSnacks — APPROVED** [review]
Should be good to go now!

File: common/src/main/java/me/shedaniel/architectury/event/events/ChunkEvent.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    
+    interface LoadData {
+        /**
+         * Invoked just before a chunk's data is fully read.
+         * You can read out your own data from the {@link CompoundTag} parameter, when you have saved one before.
+         * Equivalent to Forge's {@code ChunkDataEvent.Load}.
+         *
+         * @param chunk The chunk that is loaded.
+         * @param level The level the chunk is in. Can be {@code null}, e.g. if {@code chunk} is a {@link ProtoChunk}.
```
Comment (shedaniel) [inline common/src/main/java/me/shedaniel/architectury/event/events/ChunkEvent.java]:
```suggestion
         * @param level The level the chunk is in, may be {@code null}.
```

**Pull request merged**